### PR TITLE
Pin all lower-level libs to latest versions

### PR DESCRIPTION
**Pull request by danielsdeleo**

### CODE REVIEW

File: omnibus/config/projects/chef.rb [hunk 1]
```diff
@@ -64,6 +64,23 @@
...
+override :"util-macros",      version: "1.19.0"
+override :"pkg-config-lite",  version: "0.28-1"
+override :"makedepend",       version: "1.0.5"
+override :"libffi",           version: "3.2.1"
+override :"libyaml",          version: "0.1.6"
+override :"libiconv",         version: "1.14"
+override :"liblzma",          version: "5.2.2"
+override :"libxml2",          version: "2.9.3"
+override :"libxslt",          version: "1.1.28"
+#override :"cacerts",                             # probably best to float?
```
Comment (danielsdeleo) [inline omnibus/config/projects/chef.rb]: I think we usually set this to latest in omnibus-sw, is it worth pinning it?

File: omnibus/config/projects/chef.rb [hunk 2]
```diff
@@ -64,6 +64,23 @@
...
+#override :"openssl"                              # leave this?
+#override :"libtool"                              # according to comment in omnibus-sw, latest versions don't work on solaris
```
Comment (danielsdeleo) [inline omnibus/config/projects/chef.rb]: https://github.com/chef/omnibus-software/blob/aefb7e79d29ca746c3f843673ef5e317fa3cba54/config/software/libtool.rb#L23

**Comment by jkeiser** [comment]
:+1:

**Comment by danielsdeleo** [comment]
http://manhattan.ci.chef.co/job/chef-build/264/

**Comment by danielsdeleo** [comment]
Merged as 18f0ee2354ae2539bf63956ef99a37a188ef67eb

**Pull request closed without merging**